submissions/lalalilulu/memory-game/script.js

```diff
@@ -0,0 +1,102 @@
+//start the game
+(function launchApp() {
+    const gameBoard = document.querySelector(".memory-game");
+    const fragment = document.createElement("div");
+    createImgArray().map(img => createCard(img)).forEach(card => fragment.appendChild(card));
+    gameBoard.innerHTML = fragment.innerHTML;
+})();
+
+function createImgArray() {
+    const imgArray = [];
+    for (let i = 1; i < 11; i++) {
+        imgArray.push("kitty-" + i);
+    }
+    imgArray.push(...imgArray);
+    return imgArray;
+}
+
+function createCard(image) {
+    const divCard = createElementWithClass("div", "memory-card");
+    divCard.dataset.image = image;
+    const imgFrontCard = createElementWithClass("img", "front");
+    imgFrontCard.setAttribute("src", "img/" + image + ".png");
+    const imgBackCard = createElementWithClass("img", "back");
+    imgBackCard.setAttribute("src", "img/back.png");
+    divCard.append(imgFrontCard, imgBackCard);
+    divCard.style.order = Math.floor(Math.random() * 100);
+    return divCard;
+}
+
+function createElementWithClass(tagName, cssClass) {
+    const element = document.createElement(tagName);
+    element.classList.add(cssClass);
+    return element;
+}
+
+const cards = document.querySelectorAll(".memory-card");
+cards.forEach(card => card.addEventListener("click", flipCard));
```

Comment:
use event delegation, please. more info you can find [here](https://javascript.info/event-delegation)